saleor/graphql/product/tests/queries/test_product_query.py

```diff
@@ -1970,3 +1970,313 @@ def test_query_product_media_for_federation(
             "url": f"http://{site_settings.site.domain}/media/products/product.jpg",
         }
     ]
+
+
+@pytest.mark.parametrize(
+    "variant_id, sku, result",
+    ((False, "123", "123"), (True, None, "123")),
+)
+def test_product_variant_field_filtering(
+    staff_api_client,
+    product,
+    variant_id,
+    sku,
+    result,
+    channel_USD,
+):
+    # given
+    query = """
+    query Product($id: ID!, $channel: String, $variant_id: ID, $sku: String){
+        product(id: $id, channel: $channel){
+           variant(id: $variant_id, sku: $sku){
+            id
+            sku
+           }
+        }
+    }
+    """
+    variant = product.variants.first()
+    variables = {
+        "id": graphene.Node.to_global_id("Product", product.pk),
+        "variant_id": (
+            graphene.Node.to_global_id("ProductVariant", variant.pk)
+            if variant_id
+            else None
+        ),
+        "sku": sku,
+        "channel": channel_USD.slug,
+    }
+
+    # when
+    response = staff_api_client.post_graphql(query, variables)
+
+    # then
+    content = get_graphql_content(response)
+    assert content["data"]["product"]["variant"]["sku"] == result
+
+
+def test_product_variant_field_filtering_null_response(
+    staff_api_client,
+    product,
+    channel_USD,
+):
+    # given
+    query = """
+    query Product($id: ID!, $channel: String, $variant_id: ID, $sku: String){
+        product(id: $id, channel: $channel){
+           variant(id: $variant_id, sku: $sku){
+            id
+            sku
+           }
+        }
+    }
```

Comment:
You could create the const from that as it is done in other tests, not copy it in multiple tests.